src/test/java/testsuite/integration/container/AuroraMysqlPerformanceIntegrationTest.java

```diff
@@ -0,0 +1,226 @@
+package testsuite.integration.container;
+
+import com.mysql.cj.conf.PropertyKey;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.sql.*;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class AuroraMysqlPerformanceIntegrationTest extends AuroraMysqlIntegrationBaseTest {
+
+  private static final int REPEAT_TIMES = 5;
+  private static final List<PerfStat> enhancedFailureMonitoringPerfDataList = new ArrayList<>();
+
+  @BeforeAll
+  public static void setUp() throws IOException, SQLException {
+    AuroraMysqlIntegrationBaseTest.setUp();
+  }
+
+  @AfterAll
+  public static void cleanUp() throws IOException {
+    doWritePerfDataToFile("./build/reports/tests/FailureDetectionResults_EnhancedMonitoring.xlsx", enhancedFailureMonitoringPerfDataList);
+  }
+
+  private static void doWritePerfDataToFile(String fileName, List<PerfStat> dataList) throws IOException {
+    try (XSSFWorkbook workbook = new XSSFWorkbook()) {
+      // Title
+      final XSSFSheet sheet = workbook.createSheet("FailureDetectionResults");
+
+      // Header
+      Row row = sheet.createRow(0);
+      Cell cell = row.createCell(0);
+      cell.setCellValue("FailureDetectionGraceTime");
+      cell = row.createCell(1);
+      cell.setCellValue("FailureDetectionInterval");
+      cell = row.createCell(2);
+      cell.setCellValue("FailureDetectionCount");
+      cell = row.createCell(3);
+      cell.setCellValue("NetworkOutageDelayMillis");
+      cell = row.createCell(4);
+      cell.setCellValue("MinFailureDetectionTime");
+      cell = row.createCell(5);
+      cell.setCellValue("MaxFailureDetectionTime");
+      cell = row.createCell(6);
+      cell.setCellValue("AvgFailureDetectionTime");
```

Comment:
```suggestion
      final String[] headers = new String[]{
          "FailureDetectionGraceTime",
          "FailureDetectionInterval",
          "FailureDetectionCount",
          "NetworkOutageDelayMillis",
          "MinFailureDetectionTime",
          "MaxFailureDetectionTime",
          "AvgFailureDetectionTime"
      };

      // Header
      Row row = sheet.createRow(0);
      Cell cell;
      for (int i = 0; i < headers.length; i++) {
        cell = row.createCell(i);
        cell.setCellValue(headers[i]);
      }
```

Reply:
I'm not sure if this changes make the code more readable.

Reply:
We can hide it in a method. The original method drives me crazy.

Reply:
`buildCSVHeaders(String[] headers)`?